workflow-templates/job.build.yml

```
@@ -0,0 +1,63 @@
+name: Build Docker Image with Kaniko
+
+on:
+  workflow_call:
+    inputs:
+      aws_region:
+        required: true
+        type: string
+      service:
+        required: true
+        type: string
+      image_tag:
+        required: true
+        type: string
+      dockerfile:
+        required: false
+        type: string
+        default: Dockerfile
+      build_args:
+        required: false
+        type: string
+      context:
+        required: false
+        type: string
+        default: "."
+      cache:
+        required: false
+        type: boolean
+        default: false
+        
+jobs:
+  build-image-kaniko:
+    name: Docker Build
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: read
+
```

We will need id-token for JWT auth
```suggestion
      packages: read
      id-token: write

```